app/models/ems_azure.rb

```diff
@@ -29,17 +30,16 @@ def connect(options = {})
     self.class.raw_connect(clientid, clientkey, tenant_id)
   end
 
-  def verify_credentials(_auth_type = nil, _options = {})
-    # TODO
-    true
-  end
-
-  def tenant_id=(tenant_id)
-    self.uid_ems = tenant_id
-    save
-  end
+  def verify_credentials(*)
+    begin
+      connect
```

Comment:
No the options parameter in connect defaults to nil
The username and password come from the authentications table
tenant_id is an alias_attribute defined at the top